agents/ls-csharp/src/main/java/org/eclipse/che/api/agent/LSCSharpAgent.java

```diff
@@ -0,0 +1,108 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2017 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.api.agent;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+
+import org.eclipse.che.api.agent.shared.dto.AgentDto;
+import org.eclipse.che.api.agent.shared.model.Agent;
+import org.eclipse.che.api.core.model.workspace.ServerConf2;
+import org.eclipse.che.commons.lang.IoUtil;
+import org.eclipse.che.dto.server.DtoFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.Map;
+
+import static java.lang.String.format;
+import static java.util.Collections.unmodifiableList;
+import static java.util.Collections.unmodifiableMap;
+
+/**
+ * Language server C# agent.
+ *
+ * @see Agent
+ *
+ * @author Anatolii Bazko
+ */
+@Singleton
+public class LSCSharpAgent implements Agent {
```

Comment:
Looks like all the agents are very similar. Please consider creation of abstract base class for them.